### DISCUSSION TITLE
release 1.2.3 or 1.3.0 in March

### DISCUSSION THREAD

**Discussion by newville** [lmfit/lmfit-py#943] [General]
Merging #941 and #942, I think we're approximately ready for a release -- it's been too long. 

Are there other outstanding issues with code, documentation, or examples that we should try to squeeze in?

While reviewing the changes so far, and also #941, which is kind of a "behavior change", I think we can also ask whether this should be called 1.2.3 or 1.3.0?  I have no strong opinion.

Maybe we can aim for March 15 or April 1?

**Comment by newville** [comment]
If I don't hear objections, I'll work on a ChangeLog to tag the current release (or, approximately so) as version 1.2.3.   

Are there any other suggestions for changes to code, examples, project infrastructure, or docs?

**Reply by reneeotten** [reply]
if you haven't had the chance to work on this I might be able to do so sometime this week so that we can get a release out soon-ish? Let me know as I would rather not duplicate the work ;)

**Comment by newville** [comment]
@reneeotten Yeah, non-coding work has been taking up all my time at work the past few weeks, and the `uncertainties` project and other projects are causing a lot of traffic.  Well, and it didn't seem like a huge rush ;). 

 But, I have this about half done -- I think I can get it done this week.  FWIW, the rush at work should take a break in the next week or two..

**Comment by eendebakpt** [comment]
I tested lmfit-py locally with numpy2.0.0rc1 and all tests pass. See https://github.com/numpy/numpy/issues/24300#issuecomment-2030603395.

So if a release goes out before numpy 2.0 is released that would be great!

**Comment by newville** [comment]
@reneeotten Here is a draft changelog. Any suggestions?  I guess we could always alter it later.

Changelog:

New features:

 - add 'min_rel_change' as an optional variable in the calculation of confidence intervals with
   `conf_interval()`. (#937).

 - `Model.eval_uncertainty` now takes an optional `dscale` parameter (default value of 0.01) to
   set the step size for calculating derivatives (#933).
 - add calculation of `predicted_interval` to `Model.eval_uncertainty` (#933).

Bugfixes/enhancements:

 - restore best-fit parameter values for high accuracy values of constrained values (PR #907)
 - improvement to Model for the difference between "variable parameter", "independent variable", and "option".
   With this change, keyword arguments to model functions with non-numerice default values such as
   `do_thing=True`, or `form='linear'` has those arguments become clearly identified as independent variables,
   and use the provided values as default values. (#941)
 - the `dill` package is now required. (#940)
 - the `numexpr` is no longer required.
 - better saving/loading saved states of Model now use dill, have several cleanups, and are now versioned for
   future-proofing. Also, propagate funcdets for Parameters when loading a Model. (#932, #934)

 - adding an example using basinhopping, can take other methods as command-line argument

 - in the TNC method, `maxfun` is used instead of `maxiter`.
 - fix broken links in the Examples gallery
 - fix inter-sphinx mapping to scipy docs.
 - updates for compatibility with numpy 2.0
 - fix tests for Python 3.12 and Python 3.13-dev
 - build swicthes to pyproject.toml (#928)
 - drop support for Python 3.7, increase minimum numpy verstio to 1.23 and scipy to 1.8 (#927)

 - fix bug calculating r-squared for fits with weights (#921, #923)
 - fix for `modelresult.eval_uncertainty()` after `load_modelresult` (#909)
 - use StringIO for `pandas.read_json`.
 - add test for MinimizerResult.uvars after successful fit (#913)

**Reply by reneeotten** [reply]
that looks good to me, without going through the PR list myself ;) 

You can remove the line:
```the numexpr is no longer required.``` - this was never a requirement and only (incorrectly) snug in at some point after the previous release, so this was never something the end-user observed.

Some aspects should probably be moved to the category ```Maintenance/Deprecations:``` as we usual have in the `whatsnew.rst` file. Other than that this looks fine and we can indeed always fine-tune it in the documentation if needed. 

I would prefer to release this as version `1.3.0` given the changes, but I'll happily leave it up to you.

**Comment by newville** [comment]
@reneeotten Thanks - I agree with all of that, and will do.

**Comment by newville** [comment]
OK, I updated the `whatsnew.rst`, tagged 1.3.0, pushed to GitHub, and to PyPI.  I rebuilt the docs, including PDF, EPUB, and zipped-html.  I pushed that to gh-pages.  I'm not sure why, but the https://lmfit.github.io/lmfit-py/ has not yet been updated from that yet (even after 2+ hours, and refreshing).  I hope that happens soon!
